sunpy/net/scraper.py

```diff
@@ -72,24 +102,55 @@ class Scraper:
     pattern looks with the actual time.
     """
 
-    def __init__(self, pattern, **kwargs):
-        pattern = pattern.format(**kwargs)
-        timepattern = pattern
-        for k, v in TIME_CONVERSIONS.items():
-            if k in timepattern:
-                timepattern = timepattern.replace(k,v)
-        self.timepattern = timepattern
-        if "year:4d" in pattern and "year:2d" in pattern:
-            pattern = pattern.replace("year:2d", ":2d")
-        self.pattern = pattern
-        self.domain = f"{urlsplit(self.pattern).scheme}://{urlsplit(self.pattern).netloc}/"
-        milliseconds = re.search(r'\%e', self.timepattern)
-        if not milliseconds:
-            self.now = datetime.now().strftime(self.timepattern)
+    def __init__(self, pattern=None, regex=False, format=None, **kwargs):
+        if pattern is not None and format is None:
+            warnings.warn(
+                "Using 'pattern' for the URL format is deprecated. "
+                "Please use 'format' instead.",
+                DeprecationWarning,
+            )
+            self.use_old_format = True
+            self.pattern = pattern
+        elif pattern is None and format is not None:
+            self.use_old_format = False
+            self.pattern = format
```

Comment:
So if both are set, it will use format and ignore the pattern?

I would rather this errored and told users to only format.

Reply:
i think the else condition afterwards would catch that instead and raise the error?

Reply:
We should check with a unit test.